File: timelineviewv2/src/main/java/ro/dobrescuandrei/timelineviewv2/OnDateTimeIntervalChangedEventsLock.java
package ro.dobrescuandrei.timelineviewv2;

public class OnDateTimeIntervalChangedEventsLock
{
    volatile boolean isLocked = false;

    public synchronized void lock()
    {
        this.isLocked=true;
    }

    public synchronized void unlock()
    {
        this.isLocked=false;
    }
}

File: timelineviewv2/src/main/java/ro/dobrescuandrei/timelineviewv2/OnDateTimeIntervalChangedEventsLock.kt
package ro.dobrescuandrei.timelineviewv2

class OnDateTimeIntervalChangedEventsLock
{
    @Volatile
    private var isLocked = false

    @Synchronized
    fun lock()
    {
        isLocked = true
    }

    @Synchronized
    fun disableInvokingEvents() = lock()

    @Synchronized
    fun unlock()
    {
        isLocked = false
    }

    @Synchronized
    fun enableInvokingEvents() = unlock()

    @Synchronized
    fun isEventInvokingEnabled() = !isLocked
}